internal/controllers/machineset/machineset_controller.go

```diff
@@ -987,6 +1047,22 @@ func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *cl
 		return ctrl.Result{}, nil
 	}
 
+	// Sort the machines from newest to oldest.
+	sort.Slice(machinesToRemediate, func(i, j int) bool {
+		return machinesToRemediate[i].CreationTimestamp.After(machinesToRemediate[j].CreationTimestamp.Time)
+	})
+
+	// At this point we have a list of machines that need to be remediated, and the number of tokens available to do so.
+	// Check if we can remediate any machines.
+	if maxInFlight <= 0 {
+		// No tokens available to remediate machines.
+		return ctrl.Result{}, nil
+	} else if len(machinesToRemediate) > maxInFlight {
+		// We have more machines to remediate than tokens available.
+		machinesToRemediate = machinesToRemediate[:maxInFlight]
```

Comment:
q: should we update machine's OwnerRemediatedCondition for the remaining machines with a message that explains that remediation is on hold because X machines are already being remediated?